# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1231921711 18000
#      Wed Jan 14 03:28:31 2009 -0500
# Node ID a835406e02dd18cd7ec66578738af6c5d29eee67
# Parent  738cb6271933f66e816114182784790c1c2db8c8
load_path::do_set: call clear after disabling add_hook

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-01-14  John W. Eaton  <jwe@octave.org>
+
+	* load-path.cc (load_path::do_set): Call do_clear after disabling
+	add_hook.
+
 2009-01-13  Jaroslav Hajek  <highegg@gmail.com>
 
 	* dirfns.cc (Freadddir): qsort -> sort.
 	* graphics.cc (maybe_set_children): Dtto.
 	* help.cc (simple_help): Dtto.
 	* input.cc (generate_possible_completions): Dtto.
 	* oct_map.cc (keys_ok): Dtto.
 	* ov-fcn-inline.cc (Finline): Dtto.
diff --git a/src/load-path.cc b/src/load-path.cc
--- a/src/load-path.cc
+++ b/src/load-path.cc
@@ -530,26 +530,26 @@ split_path (const std::string& p)
     retval.push_back (elt);
 
   return retval;
 }
 
 void
 load_path::do_set (const std::string& p, bool warn)
 {
-  do_clear ();
-
   std::list<std::string> elts = split_path (p);
 
   // Temporarily disable add hook.
 
   unwind_protect_fptr (add_hook);
 
   add_hook = 0;
 
+  do_clear ();
+
   for (std::list<std::string>::const_iterator i = elts.begin ();
        i != elts.end ();
        i++)
     do_append (*i, warn);
 
   // Restore add hook and execute for all newly added directories.
 
   unwind_protect::run ();
